File: crates/laya-parse/tests/fixtures/sample.swift
import Foundation

let maxKeys = 1024

/// Something with a key.
protocol Keyed {
    func key() -> String
}

/// A key store.
class KeyStore {
    private var map: [String: String] = [:]

    init() {
        map["a"] = "b"
    }

    func get(_ key: String) -> String {
        return map[key] ?? ""
    }

    func set(_ key: String, _ value: String) {
        map[key] = value
    }
}

struct Point {
    var x: Double
    var y: Double
}

func norm(_ p: Point) -> Double {
    return (p.x * p.x + p.y * p.y).squareRoot()
}

File: crates/laya-parse/tests/fixtures/Sample.kt
package com.example

import kotlin.math.sqrt

const val MAX_KEYS = 1024

typealias Key = String

/** A key store. */
class KeyStore(private val map: MutableMap<Key, String>) {
    fun get(key: Key): String {
        return map[key] ?: ""
    }

    fun set(key: Key, value: String) {
        map[key] = value
    }

    companion object {
        fun empty(): KeyStore = KeyStore(mutableMapOf())
    }
}

object Registry {
    val stores = mutableListOf<KeyStore>()
}

fun norm(x: Double, y: Double): Double = sqrt(x * x + y * y)
